### ISSUE TITLE
stable-diffusion-xl error

### ISSUE THREAD

**Issue by cjymz886** [GaParmar/img2img-turbo#146]
@andjoer 
Hi, thank you for your work! I have this error when training by stable-diffusion-xl-base-1.0

![Image](https://github.com/user-attachments/assets/feaecfe3-6f20-41c2-82b8-9098c1df0c31)

**Comment by andjoer** [comment]
Hi, thanks for the feedback. I need to update the requirements.txt since the hf team changed the scheduler. I hope I will have time for this soon. 

In the meantime you could manually install like this. 

```
pip install torch==2.1.2 torchvision==0.16.2 torchaudio==2.1.2 --index-url https://download.pytorch.org/whl/cu121
pip install xformers==0.0.23.post1
pip install diffusers==0.31.0
```

This worked for someone else who had the same problem a couple of days ago. I'm personally currently mainly working with sd3 due to the better autoencoder. As a side note based on my experiences so far: When you have an image to image task and you want to alter texture the skip connections of the vae might prevent this since then the vae is responsible for the fine details (however adding details like converting drawings to photos works since then the vae just decodes the new details added by the unet). SD 3 vae is good enough so that it does not need skip connections anymore.

**Comment by cjymz886** [comment]
thanks, i will try

**Comment by cjymz886** [comment]
> Hi, thanks for the feedback. I need to update the requirements.txt since the hf team changed the scheduler. I hope I will have time for this soon.
> 
> In the meantime you could manually install like this.
> 
> ```
> pip install torch==2.1.2 torchvision==0.16.2 torchaudio==2.1.2 --index-url https://download.pytorch.org/whl/cu121
> pip install xformers==0.0.23.post1
> pip install diffusers==0.31.0
> ```
> 
> This worked for someone else who had the same problem a couple of days ago. I'm personally currently mainly working with sd3 due to the better autoencoder. As a side note based on my experiences so far: When you have an image to image task and you want to alter texture the skip connections of the vae might prevent this since then the vae is responsible for the fine details (however adding details like converting drawings to photos works since then the vae just decodes the new details added by the unet). SD 3 vae is good enough so that it does not need skip connections anymore.

**Comment by cjymz886** [comment]
@andjoer 
hi, I have another issue: I want to train  blurry to clear, but I find that the training results are hard to fit and the generated results are all  blurry. I am based on pix2pix_turbo and set lambda_clipsim=0. Do you have any good suggestions for improvement

thanks.

**Comment by andjoer** [comment]
hi, its hard to diagnose without seeing the loss curves. Are you logging to wandb and could I see the l2 and lpips curves? Just one general comment: the discriminator only has a resolution of 224 pixels and lpips only of 512. I added a parameter --num_patches: if you set --num_patches=2 the discriminator and lpips receives the full image plus 2*2 patches (you can of course also choose higher numbers). An other approach would be to learn it gently and increase the blur more and more with ongoing training. It is not guaranteed that it will work since this pix2pix-turbo approach is not capable of all image to image tasks. But I would try 1) patches as mentioned 2) see if it works for small blurs. 3) if you have enough vram try with sd3.

**Comment by cjymz886** [comment]
> hi, its hard to diagnose without seeing the loss curves. Are you logging to wandb and could I see the l2 and lpips curves? Just one general comment: the discriminator only has a resolution of 224 pixels and lpips only of 512. I added a parameter --num_patches: if you set --num_patches=2 the discriminator and lpips receives the full image plus 2*2 patches (you can of course also choose higher numbers). An other approach would be to learn it gently and increase the blur more and more with ongoing training. It is not guaranteed that it will work since this pix2pix-turbo approach is not capable of all image to image tasks. But I would try 1) patches as mentioned 2) see if it works for small blurs. 3) if you have enough vram try with sd3.

Thank you very much for your reply!

I adopt some of your suggestions: 1）use stable-diffusion-3.5-medium; set num_patches=2;  Besides, only the l2 loss was used before training 1000 steps;

my main params setting:
--lambda_gan=0.5
--lambda_lpips = 5.0
--lambda_l2 = 1.0
--lambda_clipsim = 0
--resolution = 512
--train_batch_size =1
--gradient_accumulation_steps =2
--max_train_steps = 2500
--num_patches =2
--learning_rate=5e-5 

The training results on 50  pair images are as follows：

#train
![Image](https://github.com/user-attachments/assets/5311db92-1b09-43a7-a237-30ae56b9b254)

#val
![Image](https://github.com/user-attachments/assets/a1554974-755e-4634-a5b1-ddb2a1ec6aff)

![Image](https://github.com/user-attachments/assets/e85978c6-18a7-4445-af0e-3729f0b8d166)


my training goal is: 1) make the input blurry image clear; 2) If the input image has a three-dimensional or curved shape, it should be changed to a flat one

At present, the training effects of both targets are not good.  So, I have a few questions: 
1) Is there too little training data? 
2) Is pix2pix-turbo not suitable for my task? 
3) The training loss fluctuates greatly, is  learning rate  too high or there are other reasons

Could you give me some good advices, thanks
